mlos_bench/mlos_bench/environments/base_environment.py

```diff
@@ -268,6 +294,9 @@ def setup(self, tunables: TunableGroups, global_config: Optional[dict] = None) -
         _LOG.info("Setup %s :: %s", self, tunables)
         assert isinstance(tunables, TunableGroups)
 
+        # TODO: Make sure we create a context before invoking setup/run/status/teardown
+        # assert self._in_context
```

Comment:
When does this get enabled?

Reply:
today, in the upcoming PR

Reply:
absolutely! once I uncomment this, all tests break and that makes it a much bigger PR, so I've decided to split it in two